health-services/project-factory/src/server/utils/pollUtils.ts

```
@@ -0,0 +1,177 @@
+};
+
+/**
+ * Polls a function at regular intervals until a condition is met or the maximum retries are reached.
+ * @param functionToBePolledFor The function to be executed for polling.
+ * @param conditionForTermination A callback function that evaluates whether polling should stop.
+ * @param pollInterval The interval (in milliseconds) between each poll attempt. Default is 2000ms.
+ * @param maxRetries The maximum number of retries before terminating polling. Default is 50.
+ * @returns A promise that resolves with the function response if the condition is met, or rejects if retries are exhausted.
+ */
+const pollForTemplateGeneration = async (
+  functionToBePolledFor: Function,
```

Biome

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)